docker/package/tezos_baking_wizard.py

```diff
@@ -576,57 +564,79 @@ def check_baker_account(self):
             print("Its current value is", value.decode("utf-8"))
             print("With a corresponding address:", address.decode("utf-8"))
 
-            return yes_or_no("Should it be replaced with the new import? <y/N> ", "no")
+            return yes_or_no(
+                "Would you like to import a new key and replace this one? <y/N> ", "no"
+            )
+        else:
+            return True
 
     # Importing the baker key
     def import_baker_key(self):
 
         replace_baker_key = self.check_baker_account()
-        valid_choice = False
-
-        while replace_baker_key and not valid_choice:
 
-            self.query_step(key_mode_query)
+        if replace_baker_key:
+            if self.config["network"] == "mainnet":
+                key_import_modes.pop("json", None)
+
+            # We define the step here to disallow choosing json baking on mainnet
```

Comment:
I mean, I guess you could have made `key_mode_query` a function with an argument too?

Not sure if it's any better, just thinking aloud.